olive/model/config/registry.py

```diff
@@ -0,0 +1,36 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+REGISTRY = {}
+
+
+def model_handler_registry(model_type):
+    """Decorate and register all OliveModelHandler subclasses.
+
+    Args:
+        model_type (str): The model type registration name
+
+    Returns:
+        cls: The class of register.
+    """
+
+    def decorator_model_class(cls):
+        if model_type in REGISTRY:
+            raise ValueError("Cannot have two model handlers with the same name")
+
+        REGISTRY[model_type.lower()] = cls
```

Comment:
Is it possible to get model_type automatically?
`model_type = model_type or model_type[:-len("handler")]`

I am not sure if it is totally 1:1 mapping between model_type and model_type's handler, like: f"{model_type}Handler"

Reply:
The model_type name will not be same with the class by removing suffix. I don't like make the tight coupled. By using the name explicitly, we could make the class name changed at anytime without breaking the back compatibility.